### DISCUSSION TITLE
How to see datamodel value at runtime ?

### DISCUSSION THREAD

**Discussion by sfabry** [alexzhornyak/SCXML-tutorial#16] [Q&A] [answered]
Hi,

When running a scxml machine in the editor, how can we see the values of the datamodel changing over time due to state entry or transitions ?

Thanks

**Comment by alexzhornyak** [comment, marked as answer]
Hi, you can use [\<log\>](https://alexzhornyak.github.io/SCXML-tutorial/Doc/log.html) element

```xml
<data id="Var1" expr="0"/>
<!-- Display variable Var1 -->
<log label="Var1" expr="Var1"/>
```

![image](https://user-images.githubusercontent.com/18611095/158053676-7e817a55-b064-449f-964c-eb126f61c858.png)
